File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> ans;
        int count = 1;
        int prev = 0;
        ans.insert(nums.begin(), nums.end());
        for(int x : ans){
            if (ans.count(x+1)){
                count = 2;
                while(ans.count(count+x)){
                    count++;
                }
            }
            prev = max(prev, count);
        }
        return prev;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = 0
        mf = 0
        helper = {}
        ans = 0
        for r in range(len(s)):
            helper[s[r]] = 1 + helper.get(s[r], 0)
            mf = max(mf, helper[s[r]])
            while ((r-left+1)-mf) > k: 
                helper[s[left]]-=1
                left+=1
                
            ans = max(ans, r - left +1)
        return ans
